File: Minesweeper_Java/src/RandomAI.java
package src;

import src.Action.ACTION;

public class RandomAI extends AI {
	private final int ROW_DIMENSION;
	private final int COL_DIMENSION;
	private final int TOTAL_MINES;
	private int flagsLeft;
	public RandomAI(int rowDimension, int colDimension, int totalMines) {
		this.ROW_DIMENSION = rowDimension;
		this.COL_DIMENSION = colDimension;
		this.TOTAL_MINES = this.flagsLeft = totalMines;
	}
	
	// Make 3 random moves and then leave
	int count = 3;
	ACTION actions[] = ACTION.values();
	// --------------------- Implement getAction() --------------------
	public Action getAction(int number) {
		int idx, x, y;
		while (count > 0) {
			count--;
			// Get a random action that is not LEAVE
			idx = (int)(Math.random() * (actions.length-1)) + 1;
			ACTION action = actions[idx];
			
			x = (int)(Math.random() * this.ROW_DIMENSION) + 1;
			y = (int)(Math.random() * this.COL_DIMENSION) + 1;
			
			return new Action(action,x,y);
		}
		return new Action(ACTION.LEAVE);	
	}
	
}

File: Minesweeper_Python/src/RandomAI.py
# ==============================CS-199==================================
# FILE:			RandomAI.py
#
# AUTHOR: 		Justin Chung
#
# DESCRIPTION:	This file contains the RandomAI class. In this class,
#				the RandomAI agent will return a random move at every turn
#				of the game.
#
# NOTES: 		- DO NOT MAKE CHANGES TO THIS FILE.
# ==============================CS-199==================================

import random
from AI import AI
from Action import Action


class RandomAI ( AI ):
	
	def __init__(self, rowDimension, colDimension, totalMines, startX, startY):
		self.__rowDimension = rowDimension
		self.__colDimension = colDimension
		self.__moveCount = 0


	def getAction(self, number: int) -> "Action Object":
		while self.__moveCount < 5:
			action = AI.Action(random.randrange(1, len(AI.Action)))
			x = random.randrange(self.__colDimension)
			y = random.randrange(self.__rowDimension)
			self.__moveCount += 1
			return Action(action, x, y)

		action = AI.Action(random.randrange(len(AI.Action)))
		x = random.randrange(self.__colDimension)
		y = random.randrange(self.__rowDimension)

		return Action(action, x, y)